winterfell/middlewares/auth.js

```diff
@@ -0,0 +1,23 @@
+var http = require('http-status-codes');
+
+function Auth(app) {
+  this.app = app;
+}
+
+module.exports = Auth;
+
+module.exports.authenticatedOr404 = function (req, res, next) {
+  if (req.session.userId) {
+    return next();
+  }
+
+  console.log("Not authed");
+  res.sendStatus(http.NOT_FOUND);
+};
+
+module.exports.authenticatedOrRedirect = function (req, res, next) {
+  if (req.session.userId)
+    return next();
+
+  res.redirect('/');
+};
```

Comment:
nit: extra line at end of file please
Sorry I'm such a 👿 
http://stackoverflow.com/questions/729692/why-should-text-files-end-with-a-newline